### ISSUE TITLE
I hope to be able to filter artist names when retrieving tags from image websites like Danbooru 

### ISSUE THREAD

**Issue by wochenlong** [deepghs/waifuc#42]
I encountered this issue while training a multi-character LORA. I would like to use the DanbooruSource to retrieve Danbooru images while preserving the original tags but filtering out the artist tags.

Because artist name tags are unfamiliar vocabulary for the model, some character features might inadvertently learn from the artist tags. Additionally, including artist name tags in the dataset could potentially lead to copyright disputes and unnecessary misunderstandings.

Here is the code I am currently using to retrieve images:
```python
from waifuc.action import *
from waifuc.export import *
from waifuc.source import *

if __name__ == "__main__":
    s = DanbooruSource(["(honkai:_star_rail"])
    s.attach(
        ModeConvertAction("RGB", "white"),
        NoMonochromeAction(),
        ClassFilterAction(["illustration", "bangumi"]),
        AlignMinSizeAction(1024),
        FilterSimilarAction(),
        RandomFilenameAction(),
        FirstNSelectAction(10000),
        FileExtAction(ext=".webp", quality=90),
    ).export(TextualInversionExporter(r"/root/data/a3/honkai:_star_rail/1_all"))
```
This is an example of the folder structure of the images I obtained:
![D782BA7A4ACA4446D12F44396B133103](https://github.com/deepghs/waifuc/assets/117965575/641e6c57-5be6-4f5a-8ab0-1877f83fb7fc)
Taking one of the images as an example:
![00f1c2c35dcdde36cd7c6da736c99744c2f923ac](https://github.com/deepghs/waifuc/assets/117965575/bd160750-3d87-4b86-adbf-ed5fdbbf679d)
Its tags are：
```
1girl, 1other, upper body, smile, streaked hair, white hair, cleavage cutout, black gloves, chibi, coin, gold, honkai: star rail, honkai \(series\), jiu fanglianhua, looking at viewer, numby \(honkai: star rail\), simple background, topaz \(honkai: star rail\), trotter \(honkai: star rail\), white background
```
Among them, **jiu_fanglianhua** is the artist name tag (Artist) that I need to filter.
The link to this image on DanbooruSource is: https://danbooru.donmai.us/posts/6596530?q=jiu_fanglianhua+

我是在训练多角色lora时遇到这个问题的。
我希望使用DanbooruSource获取Danbooru图片时，可以保留原有的tag，但是过滤掉艺术家标签。
因为艺术家名字标签对于模型来说，是陌生的词汇，部分角色的特征可能会学到艺术家标签里面去。
另外，在数据集中加入艺术家名字标签，使用艺术家名字，可能会有版权纠纷，引起一些不好的误会。
这是我现在使用获取图片的代码：
```python
from waifuc.action import *
from waifuc.export import *
from waifuc.source import *

if __name__ == "__main__":
    s = DanbooruSource(["(honkai:_star_rail"])
    s.attach(
        ModeConvertAction("RGB", "white"),
        NoMonochromeAction(),
        ClassFilterAction(["illustration", "bangumi"]),
        AlignMinSizeAction(1024),
        FilterSimilarAction(),
        RandomFilenameAction(),
        FirstNSelectAction(10000),
        FileExtAction(ext=".webp", quality=90),
    ).export(TextualInversionExporter(r"/root/data/a3/honkai:_star_rail/1_all"))
```
这个是我获取到的图片文件夹的样子：
![D782BA7A4ACA4446D12F44396B133103](https://github.com/deepghs/waifuc/assets/117965575/641e6c57-5be6-4f5a-8ab0-1877f83fb7fc)
如其中的某张图为例：
![00f1c2c35dcdde36cd7c6da736c99744c2f923ac](https://github.com/deepghs/waifuc/assets/117965575/bd160750-3d87-4b86-adbf-ed5fdbbf679d)
他的TAG为：
```
1girl, 1other, upper body, smile, streaked hair, white hair, cleavage cutout, black gloves, chibi, coin, gold, honkai: star rail, honkai \(series\), jiu fanglianhua, looking at viewer, numby \(honkai: star rail\), simple background, topaz \(honkai: star rail\), trotter \(honkai: star rail\), white background
```
其中 **jiu_fanglianhua**  是我需要过滤的艺术家名称标签（Artist）。
这张图在DanbooruSource的链接为：https://danbooru.donmai.us/posts/6596530?q=jiu_fanglianhua+

**Comment by wochenlong** [comment]
thank for your hard work !

**Comment by narugo1992** [comment]
merged #44
